### PR TITLE
Remove zcash_primitives::zip339 module

### PR DESCRIPTION
**Pull request by daira**
Supercedes #1377. This helps with Arti support in Zebra. See https://github.com/zcash/librustzcash/pull/1377#issuecomment-2098664351 for discussion.

### CODE REVIEW

File: zcash_primitives/Cargo.toml [hunk 1]
```diff
@@ -63,9 +63,6 @@ hdwallet = { workspace = true, optional = true }
 #   - `SecretKey` and `PublicKey` types exposed
 secp256k1 = { workspace = true, optional = true }
 
-# - ZIP 339
-bip0039 = { version = "0.10", features = ["std", "all-languages"] }
```
Comment (str4d) [inline zcash_primitives/Cargo.toml:67]: This removal results in changes to `Cargo.lock` that were not included in the commit. Please include those changes in the commit removing this dependency.

**Review by str4d — CHANGES_REQUESTED** [review]

File: zcash_primitives/Cargo.toml (hunk 1 above)
Comment (daira) [reply zcash_primitives/Cargo.toml:67]: Are you sure it should be in the same commit? That commit is a pure reversion of d1443e5049c32615155eaed0e36915dbe5222f6b.

Comment (str4d) [reply zcash_primitives/Cargo.toml:67]: Given that it is a pure reversion, fine to be in a follow-on commit.

**Review by str4d — APPROVED** [review]
utACK 031b416b75bb7e654504b4dbdeeec3262dfc8e10

**Comment by AArnott** [comment]
This was a breaking change for me, and IMO deserves a mention in the CHANGELOG.

**Pull request merged**